HACKING.md

```diff
@@ -0,0 +1,78 @@
+# Tips and tricks for hacking on Buck2
+
+## Building the code
+
+Buck2 is written in Rust, **and currently requires a nightly toolchain**. You
+need **[rustup](https://rustup.rs)** installed to provision the necessary
+version of `rustc` and `cargo` as a result.
+
+You can either build `buck2` from a clone of the Buck2 repo (this will cause
+rustup/cargo to install the right nightly version of `rustc`):
+
+```sh
+git clone https://github.com/facebook/buck2.git
+cd buck2/
+cargo install --path=app/buck2
+```
+
+Or alternatively install it directly from GitHub:
+
+```sh
+rustup install nightly-2023-03-07
+cargo +nightly-2023-03-07 install --git https://github.com/facebook/buck2.git buck2
+```
+
+Build uses prebuilt `protoc` binary from
+[protoc-bin-vendored](https://crates.io/crates/protoc-bin-vendored) crate. If
+these binaries to do not work on your machine, the path to `protoc` binary and
```

Comment:
Do we have a good idea where these won't work? Should we start by saying that in most cases it will work?

Reply:
Because this is a detail handled in the Nix case and should be handled on most systems with static exes, I just removed this bit.